packages/react-native-bridge/ios/GutenbergLayouts.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+public struct GutenbergPageLayouts {
+    public let layouts: [GutenbergLayout]
+    public let categories: [GutenbergLayoutCategory]
+
+    /// Contains a map of layouts based on their Category slug
+    private let groupedLayouts: [String: [GutenbergLayout]]
+
+
+    public init(layouts: [GutenbergLayout], categories: [GutenbergLayoutCategory]) {
+        self.layouts = layouts
+        self.categories = categories
+        groupedLayouts = GutenbergPageLayouts.groupLayouts(layouts)
+    }
+
+    static func groupLayouts(_ layouts: [GutenbergLayout]) -> [String:[GutenbergLayout]] {
+
+        var groupedLayouts = [String:[GutenbergLayout]]()
+
+        layouts.forEach { (layout) in
+            var group = groupedLayouts[layout.slug] ?? [GutenbergLayout]()
+            group.append(layout)
+            groupedLayouts.updateValue(group, forKey: layout.slug)
+        }
+
+        return groupedLayouts
+    }
+
+    public func layouts(forCategory slug: String) -> [GutenbergLayout] {
+        return groupedLayouts[slug] ?? []
+    }
+}
+
+public struct GutenbergLayout {
+    public let slug: String
+    public let title: String
+    public let preview: String
+    public let categories: [GutenbergLayoutCategory]
+}
+
+public struct GutenbergLayoutCategory {
+    public let slug: String
+    public let title: String
+    public let description: String
+    public let emoji: String
```

Comment:
Just thinking out loud, would it not be simpler if we had array of GutenbergLayout here instead? Otherwise we'll need to search in GutenbergLayout instances to get layouts for a specific category.